packages/react-native/src/theme/createReactNativeTheme.ts

```diff
@@ -0,0 +1,55 @@
+// Internal Style Dictionary methods
+import deepExtend from 'style-dictionary/lib/utils/deepExtend';
+import get from 'lodash/get';
+import mapValues from 'lodash/mapValues';
+
+import { defaultTheme } from './defaultTheme';
+import { ColorMode, Theme, ReactNativeTheme } from './types';
+import { Tokens } from './tokens';
+
+const mapValuesDeep = (v: object | string, callback: Function): object =>
+  typeof v == 'object'
+    ? mapValues(v, (v) => mapValuesDeep(v, callback))
+    : (callback(v) as object);
+
+/**
+ * This will be used like `const myTheme = createReactNativeTheme({})`
+ * `myTheme` can then be passed to a Provider
+ * const myTheme = createReactNativeTheme({})
+ * const myOtherTheme = createReactNativeTheme({}, myTheme);
+ */
+export const createReactNativeTheme = (
+  theme?: Theme,
+  colorMode?: ColorMode
+): ReactNativeTheme => {
+  // merge theme and defaultTheme to get a complete theme
+  // deepExtend is an internal Style Dictionary method
+  // that performs a deep merge on n objects.
+  const mergedTheme: ReactNativeTheme = deepExtend([
+    {},
+    defaultTheme,
+    theme,
+  ]) as ReactNativeTheme;
+  const { name, overrides } = mergedTheme;
+
+  // Setting up the tokens.
+  // At the end of this, each token should have a raw value
+  // All references to tokens will be replaced
+  const tokens = mapValuesDeep(mergedTheme.tokens, (value: string) => {
+    return typeof value == 'string' && value.startsWith('{')
+      ? (get(
+          mergedTheme.tokens,
+          value.substring(1, value.length - 2) // remove { }
+        ) as string)
+      : value;
```

Comment:
A few things here:
- Can we move this callback in to a strongly typed util? 
- Think we can improve the  comment regarding removing `{` and `}` from strings. Future maintainers may not understand that in some scenarios a string is being passed and parsed to find the value.
- What happens when `value.length - 2` does not evaluate to `}`? In general, how are we handling strings that are malformed in context to what the lookup syntax expects?

Reply:
Unresolving this conversation, still not unclear on:
> What happens when value.length - 2 does not evaluate to }? In general, how are we handling strings that are malformed in context to what the lookup syntax expects?

Reply:
and if the string does not start/end with curlies then it gets returned as is